### ISSUE TITLE
[pthreads] Build error VS2022 arm64-windows-static

### ISSUE THREAD

**Issue by syrel** [microsoft/vcpkg#27258]
Package: pthreads[core]:arm64-windows-static -> 3.0.0#11

**Host Environment**

- Host: arm64-windows
- Compiler: MSVC 19.34.31921.1
-    vcpkg-tool version: 2022-10-12-b586c2752f75bcc3f6a243749e9a0d94d0d93ccd
    vcpkg-scripts version: a2261f65e 2022-10-14 (2 days ago)

**To Reproduce**

`vcpkg --triplet arm64-windows-static install pthreads`

**Failure logs**

```
-- Using cached pthreads4w-code-v3.0.0.zip.
-- Extracting source C:/vcpkg/downloads/pthreads4w-code-v3.0.0.zip
-- Applying patch fix-arm-macro.patch
-- Applying patch use-mt.patch
-- Using source at C:/vcpkg/buildtrees/pthreads/src/49e541b66c-5fd69e67f1.clean
-- Building arm64-windows-static-rel
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:96 (message):
    Command failed: "C:/Program Files/Microsoft Visual Studio/2022/Preview/VC/Tools/MSVC/14.34.31921/bin/Hostarm64/arm64/nmake.exe" -f Makefile all install DESTROOT="C:\\vcpkg\\packages\\pthreads_arm64-windows-static"
    Working Directory: C:/vcpkg/buildtrees/pthreads/src/49e541b66c-5fd69e67f1.clean
    Error code: 2
    See logs for more information:
      C:\vcpkg\buildtrees\pthreads\nmake-build-arm64-windows-static-release-out.log
      C:\vcpkg\buildtrees\pthreads\nmake-build-arm64-windows-static-release-err.log

Call Stack (most recent call first):
  ports/pthreads/portfile.cmake:39 (vcpkg_execute_required_process)
  scripts/ports.cmake:147 (include)



```
<details><summary>C:\vcpkg\buildtrees\pthreads\nmake-build-arm64-windows-static-release-out.log</summary>

```
	"C:\Program Files\Microsoft Visual Studio\2022\Preview\VC\Tools\MSVC\14.34.31921\bin\Hostarm64\arm64\nmake.exe" /E clean VC-static
	if exist *.obj del *.obj
	if exist *.def del *.def
	if exist *.ilk del *.ilk
	if exist *.pdb del *.pdb
	if exist *.exp del *.exp
	if exist *.map del *.map
	if exist *.o del *.o
	if exist *.i del *.i
	if exist *.res del *.res
	if exist *_stamp del *_stamp
	cd tests && "C:\Program Files\Microsoft Visual Studio\2022\Preview\VC\Tools\MSVC\14.34.31921\bin\Hostarm64\arm64\nmake.exe" clean
	if exist *.dll erase *.dll
	if exist *.lib erase *.lib
	if exist _ptw32.h erase _ptw32.h
	if exist pthread.h erase pthread.h
	if exist semaphore.h erase semaphore.h
	if exist sched.h erase sched.h
	if exist *.e erase *.e
	if exist *.i erase *.i
	if exist *.obj erase *.obj
	if exist *.pdb erase *.pdb
	if exist *.o erase *.o
	if exist *.asm erase *.asm
	if exist *.exe erase *.exe
	if exist *.manifest erase *.manifest
	if exist *.pass erase *.pass
	if exist *.bench erase *.bench
	cl /errorReport:none /nologo /MT /I. /DHAVE_CONFIG_H /W3 /O2 /Ob2  /D__PTW32_STATIC_LIB /D__PTW32_BUILD_INLINED /D__PTW32_CLEANUP_C /Fopthread.o -c pthread.c
pthread.c
	rc /d__PTW32_ARCHARM64 /d__PTW32_RC_MSC /d__PTW32_CLEANUP_C version.rc
Microsoft (R) Windows (R) Resource Compiler Version 10.0.10011.16384

Copyright (C) Microsoft Corporation.  All rights reserved.


version.rc(132) : error RC2104 : undefined keyword or key name: __PTW32_VERSIONINFO_DESCRIPTION

```
</details>
<details><summary>C:\vcpkg\buildtrees\pthreads\nmake-build-arm64-windows-static-release-err.log</summary>

```

Microsoft (R) Program Maintenance Utility Version 14.34.31921.1
Copyright (C) Microsoft Corporation.  All rights reserved.


Microsoft (R) Program Maintenance Utility Version 14.34.31921.1
Copyright (C) Microsoft Corporation.  All rights reserved.


Microsoft (R) Program Maintenance Utility Version 14.34.31921.1
Copyright (C) Microsoft Corporation.  All rights reserved.

NMAKE : fatal error U1077: '"C:\Program Files (x86)\Windows Kits\10\bin\10.0.22621.0\\arm64\rc.EXE"' : return code '0x1'
Stop.
NMAKE : fatal error U1077: '"C:\Program Files\Microsoft Visual Studio\2022\Preview\VC\Tools\MSVC\14.34.31921\bin\Hostarm64\arm64\nmake.exe"' : return code '0x2'
Stop.
NMAKE : fatal error U1077: '"C:\Program Files\Microsoft Visual Studio\2022\Preview\VC\Tools\MSVC\14.34.31921\bin\Hostarm64\arm64\nmake.exe"' : return code '0x2'
Stop.
```
</details>

**Comment by Cheney-W** [comment]
Is this fix solution from upstream?

**Comment by syrel** [comment]
Hi @Cheney-W.
No, the latest release of pthreads (v 3.0.0) was in 2018. Seems like this has to be made into a patch that changes `version.rs`, just like other arm fixes e.g. `fix-arm-macro.patch`.

**Comment by JonLiu1993** [comment]
Fixed by PR #27299.
